crates/bevy_ecs/src/result/assume.rs

```diff
@@ -0,0 +1,59 @@
+use super::Error;
+
+/// Assume that `Self<T>` is `T`, otherwise return the provided error.
```

Comment:
This needs more context on why this might be useful, and the fact that it replaces `unwrap`.